runtime/pkg/rilltime/rilltime.go

```diff
@@ -0,0 +1,394 @@
+		{"Grain", `[smhdDWQMY]`},
+		// this has to be at the end
+		{"TimeZone", `{.+?}`},
+		{"AbsoluteTime", `\d{4}-\d{2}-\d{2} \d{2}:\d{2}`},
+		{"AbsoluteDate", `\d{4}-\d{2}-\d{2}`},
+		{"Number", `[-+]?\d+`},
+		// needed for misc. direct character references used
+		{"Punct", `[-[!@#$%^&*()+_={}\|:;"'<,>.?/]|]`},
+		{"Whitespace", `[ \t\n\r]+`},
+	})
+	daxNotations = map[string]string{
+		// Mapping for our old rill-<DAX> syntax
+		"TD":  "0d,latest",
+		"WTD": "0W,latest",
+		"MTD": "0M,latest",
+		"QTD": "0Q,latest",
+		"YTD": "0Y,latest",
+		"PDC": "-1d,0d",
+		"PWC": "-1W,0W",
+		"PMC": "-1M,0M",
+		"PQC": "-1Q,0Q",
+		"PYC": "-1Y,0Y",
+		// TODO: previous period is contextual. should be handled in UI
+		"PP": "",
+		"PD": "-1d,0d",
+		"PW": "-1W,0W",
+		"PM": "-1M,0M",
+		"PQ": "-1Q,0Q",
+		"PY": "-1Y,0Y",
+	}
+	rillTimeParser = participle.MustBuild[Expression](
+		participle.Lexer(rillTimeLexer),
+		participle.Elide("Whitespace"),
+	)
+	grainMap = map[string]timeutil.TimeGrain{
+		"s": timeutil.TimeGrainSecond,
+		"m": timeutil.TimeGrainMinute,
+		"h": timeutil.TimeGrainHour,
+		"d": timeutil.TimeGrainDay,
+		"D": timeutil.TimeGrainDay,
+		"W": timeutil.TimeGrainWeek,
+		"Q": timeutil.TimeGrainQuarter,
+		"M": timeutil.TimeGrainMonth,
+		"Y": timeutil.TimeGrainYear,
+	}
+)
+
+type Expression struct {
+	Start         *TimeAnchor  `parser:"  @@"`
+	End           *TimeAnchor  `parser:"(',' @@)?"`
+	Modifiers     *Modifiers   `parser:"(':' @@)?"`
+	AtModifiers   *AtModifiers `parser:"('@' @@)?"`
+	isNewFormat   bool
+	grain         *Grain
+	truncateGrain timeutil.TimeGrain
+	isComplete    bool
+	timeZone      *time.Location
+}
+
+type TimeAnchor struct {
+	Grain       *Grain  `parser:"( @@"`
+	AbsDate     *string `parser:"| @AbsoluteDate"`
+	AbsTime     *string `parser:"| @AbsoluteTime"`
+	Earliest    bool    `parser:"| @Earliest"`
+	Now         bool    `parser:"| @Now"`
+	Latest      bool    `parser:"| @Latest"`
+	Watermark   bool    `parser:"| @Watermark)"`
+	Offset      *Grain  `parser:"@@?"`
+	Trunc       *string `parser:"  ('/' @Grain)?"`
+	isoDuration *duration.StandardDuration
+}
+
+type Modifiers struct {
+	Grain         *Grain `parser:"( @@"`
+	CompleteGrain *Grain `parser:"| '|' @@ '|')?"`
+}
+
+type Grain struct {
+	Num   *int   `parser:"@Number?"`
+	Grain string `parser:"@Grain"`
+}
+
+type AtModifiers struct {
+	Offset   *TimeAnchor `parser:"@@?"`
+	TimeZone *string     `parser:"@TimeZone?"`
+}
+
+type EvalOptions struct {
+	Now        time.Time
+	MinTime    time.Time
+	MaxTime    time.Time
+	Watermark  time.Time
+	FirstDay   int
+	FirstMonth int
+}
+
+func Parse(from string) (*Expression, error) {
+	var rt *Expression
+	var err error
+
+	rt, err = ParseISO(from, false)
+	if err != nil {
+		return nil, err
+	}
+
+	if rt == nil {
+		rt, err = rillTimeParser.ParseString("", from)
+		if err != nil {
+			return nil, err
+		}
+		rt.isNewFormat = true
+	}
+
+	rt.timeZone = time.UTC
```

Comment:
Can we add support for passing in the default time zone? E.g. we have had some requests to use the same time zone as the `refresh:` clause when evaluating and formatting data for alerts and reports.